src/dmd/init.d

```diff
@@ -217,6 +223,36 @@ extern (C++) final class ExpInitializer : Initializer
     }
 }
 
+struct Designator
+{
+    Expression exp;     /// [ constant-expression ]
+    Identifier ident;   /// . identifier
+
+    this(Expression exp) { this.exp = exp; }
+    this(Identifier ident) { this.ident = ident; }
+}
+
+struct DesigInit
+{
+    Designators* designatorList; /// designation (opt)
+    Initializer initializer;     /// initializer
+}
+
+extern (C++) final class CInitializer : Initializer
+{
+    DesigInits initializerList; /// initializer-list
+
+    extern (D) this(const ref Loc loc)
+    {
+        super(loc, InitKind.C_);
+    }
+
+    override void accept(Visitor v)
+    {
+        v.visit(this);
+    }
+}
```

Comment:
Missing DDOC

Reply:
I know. I'm just seeing how it does at the moment.

Reply:
done